crates/forge/src/running.rs

```diff
@@ -229,35 +234,42 @@ pub(crate) fn run_test_case(
         &string_to_hint,
     );
 
-    Ok(runner.run_function(
+    match runner.run_function(
         runner.find_function(case.name.as_str())?,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
-    )?)
+    ) {
+        Ok(result) => Ok(TestCaseSummary::from_run_result(result, case, args)),
+        // CairoRunError comes from VirtualMachineError which may come from HintException that originates in the TestSyscallExecutionHandler
+        Err(RunnerError::CairoRunError(error)) => Ok(TestCaseSummary::Failed {
+            name: case.name.clone(),
+            msg: Some(format!(
+                "\n    {}\n",
+                error.to_string().replace(" Custom Hint Error: ", "\n    ")
+            )),
```

Comment:
I know we had this code for a while now, but what in case the error is something different than `Custom Hint Error`. Shouldn't we kill the runner as well, as we do with other errors?

Reply:
Not really, for example when reading from a file and the file does not exists this logic does not apply

Reply:
> I generally don't understand those classes of errors. What is RunnerError? What is StateError?

External libraries errors, decided to remove them and just make sure everything that fails in `run_from_test` results in failed test (does not kill the runner)